src/main/java/org/openstreetmap/atlas/geography/Segment.java

```diff
@@ -138,6 +142,40 @@ public Location intersection(final Segment that)
             return new Location(Latitude.degrees(p0Y + tValue * s1Y),
                     Longitude.degrees(p0X + tValue * s1X));
         }
+        return this.intersectionAtEnd(that);
+    }
+
+    /**
+     * Implements the same function as intersection, but with long to get the intersection at end of two segments
```

Comment:
Is there some reason we would not always want to use this version of the function?

Reply:
It sounds like the next step should be to confer with Jack, then.